MapboxMobileEvents/MMEMetricsManager.m

```diff
@@ -0,0 +1,160 @@
+        
+        for (MMEEvent *event in eventQueue) {
+            NSNumber *eventCount = [self.eventCountPerType objectForKey:event.name];
+            eventCount = [NSNumber numberWithInteger:[eventCount integerValue] + 1];
+            [self.eventCountPerType setObject:eventCount forKey:event.name];
+        }
+    }
+}
+
+- (void)metricsFromEvents:(nullable NSArray *)events error:(nullable NSError *)error {
+    if (error == nil) {
+        self.requests = self.requests + 1;
+    } else {
+        if (events) {
+            self.eventCountFailed = self.eventCountFailed + (int)events.count;
+        }
+        
+        if ([error.userInfo objectForKey:MMEResponseKey]) {
+            NSHTTPURLResponse *response = (NSHTTPURLResponse *)[error.userInfo objectForKey:MMEResponseKey];
+            NSString *urlString = [[response URL] absoluteString];
+            NSNumber *statusCode = [NSNumber numberWithLong:(long)response.statusCode];
+            NSString *statusCodeString = [statusCode stringValue];
+            NSString *failedRequestKey = [NSString stringWithFormat:@"%@, %@",urlString, statusCodeString];
+            
+            if (self.failedRequestsDict == nil) {
+                self.failedRequestsDict = [[NSMutableDictionary alloc] init];
+            }
+        
+            NSNumber *failedCount = [self.failedRequestsDict objectForKey:failedRequestKey];
+            failedCount = [NSNumber numberWithInteger:[failedCount integerValue] + 1];
+            [self.failedRequestsDict setObject:failedCount forKey:failedRequestKey];
+        }
+    }
+}
+
+- (void)metricsFromData:(NSData *)data {
+    self.totalDataTransfer = self.totalDataTransfer + data.length;
+    
+    if ([[MMEReachability reachabilityForLocalWiFi] isReachableViaWiFi]) {
+        self.wifiDataTransfer = self.wifiDataTransfer + data.length;
+    } else {
+        self.cellDataTransfer = self.cellDataTransfer + data.length;
+    }
+}
+
+- (void)incrementAppWakeUpCount {
+    self.appWakeups = self.appWakeups + 1;
+}
+
+- (void)captureConfigurationJSON:(NSDictionary *)configuration {
+    if (configuration) {
+        self.configResponseDict = configuration;
+    }
+}
+
+- (void)captureLatitude:(float)lat longitude:(float)lon {
+    if (!self.deviceLat || !self.deviceLon) {
+        self.deviceLat = [[NSString stringWithFormat:@"%.06f", lat] floatValue];
+        self.deviceLon = [[NSString stringWithFormat:@"%.06f", lon] floatValue];
+    }
+}
+
+- (void)updateDateUTC {
+    self.dateUTC = [NSDate date];
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    NSTimeZone *utcTimeZone = [NSTimeZone timeZoneWithName:@"UTC"];
+    [dateFormatter setTimeZone:utcTimeZone];
+    [dateFormatter setDateFormat:@"yyyy-MM-dd"];
+    self.dateUTCString = [dateFormatter stringFromDate:self.dateUTC];
+}
+
+- (void)resetMetrics {
+    self.failedRequestsDict = [[NSMutableDictionary alloc] init];
```

Comment:
Perhaps some kind of metrics object would be useful here, so you can just allocate a new one?